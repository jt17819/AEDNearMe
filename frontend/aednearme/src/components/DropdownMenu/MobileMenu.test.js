import { render, fireEvent, screen, cleanup, within } from '@testing-library/react';
import { MemoryRouter, Router as Router } from 'react-router-dom';
import ReactDOM from 'react-dom';
import { createMemoryHistory } from 'history'
import {default as MobileMenu} from './MobileMenu';
import axios from 'axios'


describe('Mobile Menu', () => {

  // beforeEach(() => {
  //   render(<App />, { wrapper: MemoryRouter })
  // });



  // Rendering

  it('renders a button', () => {
    render(<MobileMenu />, { wrapper: MemoryRouter })
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  test('list contains 1 buttons', () => {
    render(<MobileMenu />, { wrapper: MemoryRouter });
  
    // const listElement = screen.getAllByRole('button');
    const listItems = screen.getAllByRole('button');
  
    
    expect(listItems.length).toEqual(1);
  });



  // Snapshot

  it('should take a snapshot', () => {
    render(<MobileMenu />, { wrapper: MemoryRouter })
    const { asFragment } = render(<MobileMenu />, { wrapper: MemoryRouter })
    
    expect(asFragment(<MobileMenu />)).toMatchSnapshot()
  })

  

  
})

