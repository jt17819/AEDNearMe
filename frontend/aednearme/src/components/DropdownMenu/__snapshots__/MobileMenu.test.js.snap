// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile Menu should take a snapshot 1`] = `
<DocumentFragment>
  <nav
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    style="background: rgb(186, 24, 27); color: rgb(255, 255, 255);"
  >
    <div
      class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </nav>
</DocumentFragment>
`;
